js-exercises/animal-sort/animalSort.js

```diff
@@ -0,0 +1,23 @@
+const animalSort = animals => {
+  if (!(animals instanceof Array)) {
+    throw Error(`Expected Array as an input, got ${typeof animals}`);
+  }
+
+  const requiredKeys = ['numberOfLegs', 'name'];
+
+  for (const animal of animals) {
+    if (!(requiredKeys.every(key => Object.keys(animal).includes(key)))) {
+      throw Error('Expected keys : "numberOfLegs" and "name" not found.');
+    }
+  }
```

Comment:
`Object.keys(null)` will throw error here. which won't throw your custom message.
To ensure your custom message is `thrown as error` + some breathing in code, I suggest to use array utils more here

```suggestion
const isObject = (argument) => argument !== null && typeof argument === 'object';
const requiredKeysPresent  = animals.every(animal => isObject(animal) && requiredKeys.every(key => animal[key] !== undefined));

If(!requiredKeysPresent) {
 throw Error('Expected keys : "numberOfLegs" and "name" not found.');
}
```